include/eos/fitting/fitting.hpp

```diff
@@ -150,57 +150,51 @@ inline Eigen::VectorXf fit_shape(Eigen::Matrix<float, 3, 4> affine_camera_matrix
- * corresponding 3D vertex indices and returns them, along with the coordinates of the 3D points.
- *
- * The function only returns points which the landmark mapper was able to convert, and skips all
- * points for which there is no mapping. Thus, the number of returned points might be smaller than
- * the number of input points.
- * All three output vectors have the same size and contain the points in the same order.
- * \c landmarks can be an eos::core::LandmarkCollection<cv::Vec2f> or an rcr::LandmarkCollection<cv::Vec2f>.
- *
- * Notes:
- * - Split into two functions, one which maps from 2D LMs to vtx_idx and returns a reduced vec of 2D LMs.
- *   And then the other one to go from vtx_idx to a vector<Vec4f>.
- * - Place in a potentially more appropriate header (shape-fitting?).
- * - Could move to detail namespace or forward-declare.
- * - \c landmarks has to be a collection of LMs, with size(), [] and Vec2f ::coordinates.
- * - Probably model_points would better be a Vector3f and not in homogeneous coordinates?
+ * @brief Gets the vertex index from the given landmark name using the landmark mapper and if provided the landmark definitions.
  *
- * @param[in] landmarks A LandmarkCollection of 2D landmarks.
+ * The function only returns the vertex index if the landmark mapper was able to convert the name.
+ *
+ * @param[in] landmark_name Name of the landmark, often used as identifier.
  * @param[in] landmark_mapper A mapper which maps the 2D landmark identifiers to 3D model vertex indices.
- * @param[in] morphable_model Model to get the 3D point coordinates from.
- * @return A tuple of [image_points, model_points, vertex_indices].
+ * @param[in] landmark_definitions A set of landmark definitions for the model, mapping from identifiers to vertex indices.
+ * @return An optional int with the vertex index.
  */
-template <class T>
-inline auto get_corresponding_pointset(const T& landmarks, const core::LandmarkMapper& landmark_mapper,
-                                       const morphablemodel::MorphableModel& morphable_model)
+inline std::optional<int> get_vertex_index(const std::string landmark_name, const core::LandmarkMapper& landmark_mapper, 
+                            cpp17::optional<std::unordered_map<std::string, int>> landmark_definitions)
 {
-    using Eigen::Vector2f;
-    using Eigen::Vector4f;
-    using std::vector;
-
-    // These will be the final 2D and 3D points used for the fitting:
-    vector<Vector4f> model_points; // the points in the 3D shape model
-    vector<int> vertex_indices;    // their vertex indices
-    vector<Vector2f> image_points; // the corresponding 2D landmark points
-
-    // Sub-select all the landmarks which we have a mapping for (i.e. that are defined in the 3DMM):
-    for (int i = 0; i < landmarks.size(); ++i)
+    const auto converted_name = landmark_mapper.convert(landmark_name);
+    if (!converted_name)
+    { // no mapping defined for the current landmark
+        return std::nullopt;
```

Comment:
For  now we want to stick with eos's `cpp17::nullopt` for these. I think for one of the next releases, I'd like to move to C++17 as a minimum, then we can drop all the std/cpp17 "magic" for `optional`, but for now, let's keep it compatible with C++14.

Reply:
Oh I see I'll make the change now